### ISSUE TITLE
Google Social Auth with RN Lock - disllowed_useragent

### ISSUE THREAD

**Issue by steve-gray** [auth0/react-native-lock#105]
When using react-native-lock with credentials created as per the Auth0 integration guide at https://auth0.com/docs/connections/social/google#1-access-the-google-api-manager - I'm getting 'disallowed_useragent' 

I've installed lock into my project with react-native link and every other provider I've thrown at this seems to work fine, but not Google.

- Lock version - 0.4.0
- iOS 9 
- React Native & React version: 0.35
- NPM version: 3.10.3

![screen shot 2016-11-25 at 9 56 12 am](https://cloud.githubusercontent.com/assets/14563775/20611417/78d5fac6-b2f5-11e6-957b-8b5f87dc64d6.png)

**Comment by septemptus** [comment]
https://9to5google.com/2016/08/22/google-oauth-third-party-disable/ possibly related

**Comment by stami** [comment]
@hzalaz This is a real issue. Is Auth0 supporting this project?

**Comment by lukecwilliams** [comment]
@hzalaz Hi Hernan - we're considering whether to bet on Auth0 with React Native.  Can you confirm Auth0 is committed to handling this issue well before the changeover on April 20?

That would mean **the RN Lock change needs to be implemented by approx. end of this month** to give us time to update Auth0 RN applications.

Google's announcement for reference: https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html

Thanks!

**Comment by tjaartvdwalt** [comment]
any feedback on this issue?

**Comment by hzalaz** [comment]
Since I want to avoid submission of issues to invalid repos, will answer here. 
We know of the issue but we have been busy with other Lock/SDKs to work one thins one (in some of them fixing this issue). The Lock (iOS and Android) that this library wraps has support to do browser based authentication for a long time but since that required a developer who is willing to delve a bit in iOS/Android (which most react-native devs are not) we decided to make webview as default when we built the library. Now we can bridge the Lock for iOS and Lock for Android support for browser based authentication
- iOS: https://github.com/auth0/Lock.swift/tree/v1#important-google-connections
- Android: https://github.com/auth0/Lock.Android/tree/v1#requirements

Its on our backlog but we can't promise it will be available this week, maybe next one, its not hard to do so anyone willing to tackle it can fork it and switch from webview to browser.
